deepspeed/runtime/engine.py

```diff
@@ -242,6 +242,9 @@ def __init__(
                     assert not self.elasticity_enabled(), ("Elasticity is not currently supported"
                                                            " with model parallelism.")
 
+        if self.fp16_enabled() and not get_accelerator().is_fp16_supported():
```

Comment:
Can you please move this into `_do_sanity_check()`?

Reply:
@tjruwase Done